lib/cli/src/Cardano/CLI.hs

```diff
@@ -400,27 +405,38 @@ hoistKeyScheme eta s = CliKeyScheme
     { allowedWordLengths = allowedWordLengths s
     , mnemonicToRootKey = eta . mnemonicToRootKey s
     , deriveChildKey = \k i -> eta $ deriveChildKey s k i
+    , toPublic = eta . toPublic s
+    , inspect = eta . inspect s
     }
 
 -- | Pair of functions representing the bidirectional transformation between
 -- a @XPrv@ and its 96-byte hex-encoded form.
-xPrvToTextTransform :: (XPrv -> Either String Text, Text -> Either String XPrv)
-xPrvToTextTransform = (xPrvToHexText, hexTextToXPrv)
+xPrvToTextTransform :: (XPrvOrXPub -> Either String Text, Text -> Either String XPrvOrXPub)
+xPrvToTextTransform = (encode, decode)
   where
-    xPrvToHexText :: XPrv -> Either String Text
-    xPrvToHexText =
+    encode :: XPrvOrXPub -> Either String Text
+    encode (AXPub xpub) = return . T.pack . B8.unpack . hex . CC.unXPub $ xpub
+    encode (AXPrv xprv) =
         fmap (T.pack . B8.unpack . hex)
         . left showErr
-        . unXPrvStripPubCheckRoundtrip
+        . unXPrvStripPubCheckRoundtrip $ xprv
       where
         -- NOTE: This error should never happen from using the CLI.
         showErr ErrCannotRoundtripToSameXPrv =
             "Internal error: Failed to safely encode an extended private key"
 
-    hexTextToXPrv :: Text -> Either String XPrv
-    hexTextToXPrv txt = do
+    decode :: Text -> Either String XPrvOrXPub
+    decode txt = do
         bytes <- fromHex $ B8.pack $ T.unpack . T.strip $ txt
-        left showErr $ xPrvFromStrippedPubXPrvCheckRoundtrip bytes
+        case BS.length bytes of
+            96 -> fmap AXPrv $ left showErr $ xPrvFromStrippedPubXPrvCheckRoundtrip bytes
+            64 -> fmap AXPub . CC.xpub $ bytes
```

Comment:
Should use named constant

Reply:
This function really looks like it should be two functions :no_mouth: ...

Reply:
See for instance:

- https://input-output-hk.github.io/cardano-addresses/haddock/Cardano-Address-Derivation.html#v:xpubFromBytes
- https://input-output-hk.github.io/cardano-addresses/haddock/Cardano-Address-Derivation.html#v:xprvFromBytes

Reply:
I intentionally wanted a single coder for `XPrvOrXPub` instead of ones for `XPub` and `XPrv` individually.

This way the "is the key private or public?" question `key child` has to answer can be handled by the coder and tested via roundtrips.

Reply:
Or are you referring to the fact that it is written as a tuple `(encode, decode)`?

I wanted something more opaque than
```
mapKey (encodeHex, decodeHex) scheme
```

since it is important that the transformation is bidirectional

That is why it was called:
```
mapKey xPrvToTextTransform scheme
```

need to rename it now, maybe
```
mapKey hexKeyEncoding scheme
```
or something 🤷‍♂️

Reply:
I don't know, it feels wrong to have a single function that decodes an xpub or xprv, I'd expect two here since, whether you want one or the other is clear from the context where these functions are called. Although I get your point, if we want one common interface where we can just call `cardano-wallet key child` irrespective of whether the key is private or public then, it's not clear from the surrounding context anymore which decoder you should call. 

Still, I'd find it nicer to go for an alternative, trying one, and then the other.

Reply:
> Still, I'd find it nicer to go for an alternative, trying one, and then the other.

`<|>` gives you less control over error messages, so I'm trying to avoid it.

>  I'd expect two here since

Well, not needed. Unless we differentiated between `xprv xpub any` and had separate encodings for all three. This way `key public` could avoid suggesting public keys are accepted 
```
$ cardano-wallet-byron key public 0000
Expected key to be 96 bytes in the case of a private key and, 64 bytes for public keys. This key is 2 bytes.
```
but then not:
```
$ cardano-wallet-byron key public 00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
Input is already a public key.
```

But since it is only one command, and fairly self-explanatory I thought it would be overkill.